src/Microsoft.Azure.ServiceBus/SessionReceivePump.cs

```diff
@@ -323,7 +314,7 @@ void RaiseExceptionRecieved(Exception e, string action)
                     // Lets not bother user with this exception.
                     if (!(exception is TaskCanceledException))
                     {
-                        this.sessionHandlerOptions.RaiseExceptionReceived(new ExceptionReceivedEventArgs(exception, ExceptionReceivedEventArgsAction.RenewLock));
+                        this.sessionHandlerOptions.RaiseExceptionReceived(new ExceptionReceivedEventArgs(exception, ExceptionReceivedEventArgsAction.RenewLock, namespaceName));
```

Comment:
I think there are other places which you need to cover. For instance `CloseSessionIfNeededAsync()` method. Do take a second look at the usage of `RaiseExceptionReceived`